components/admin_console/group_settings/group_details/group_details.jsx

```diff
@@ -0,0 +1,220 @@
+                    subtitleId={t('admin.group_settings.group_detail.groupProfileDescription')}
+                    subtitleDefaultMessage='The name for this group.'
+                >
+                    <GroupProfile
+                        name={group.display_name}
+                    />
+                </AdminPanel>
+
+                <AdminPanel
+                    id='group_teams_and_channels'
+                    titleId={t('admin.group_settings.group_detail.groupTeamsAndChannelsTitle')}
+                    titleDefaultMessage='Team and Channel Membership'
+                    subtitleId={t('admin.group_settings.group_detail.groupTeamsAndChannelsDescription')}
+                    subtitleDefaultMessage='Set default teams and channels for group members. Teams added will include default channels, town-square, and off-topic. Adding a channel without setting the team will add the implied team to the listing below, but not to the group specifically.'
+                    action={(
+                        <div className='group-profile-add-menu'>
+                            <button
+                                className='btn btn-primary'
+                                onClick={this.toggleAddTeamOrChannel}
+                            >
+                                <FormattedMessage
+                                    id='admin.group_settings.group_details.add_team_or_channel'
+                                    defaultMessage='Add Team or Channel'
+                                />
+                                <i className={'fa fa-caret-down'}/>
+                            </button>
+                            {this.state.addTeamOrChannelOpen && (
+                                <ul className='add-team-or-channel-menu'>
+                                    <a onClick={this.openAddTeam}>
+                                        <FormattedMessage
+                                            id='admin.group_settings.group_details.add_team'
+                                            defaultMessage='Add Team'
+                                        />
+                                    </a>
+                                    <a onClick={this.openAddChannel}>
+                                        <FormattedMessage
+                                            id='admin.group_settings.group_details.add_channel'
+                                            defaultMessage='Add Channel'
+                                        />
+                                    </a>
+                                </ul>
+                            )}
+                        </div>
+                    )}
+                >
+                    <GroupTeamsAndChannels
+                        id={this.props.groupID}
+                        teams={groupTeams}
+                        channels={groupChannels}
+                        loading={this.state.loadingTeamsAndChannels}
+                        actions={{
```

Comment:
`TeamSelectorModal` and `ChannelSelectorModal` aren't pure anyway, and I don't worried about the performance there. I think I'll keep this as is now in the sake of clarity. I changed the other because they are pure and easy to change.

Reply:
Fixed the other case, leaved the Team and Channel selector as is.